StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/DiagnosticVerifier.cs

```diff
@@ -135,71 +135,142 @@ private static void VerifyDiagnosticResults(IEnumerable<Diagnostic> actualResult
-                        string.Format("Expected diagnostic id to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
-                            expected.Id, actual.Id, FormatDiagnostics(analyzer, actual)));
+                        string.Format("Expected:\nA project diagnostic with No location\nActual:\n{0}",
+                        FormatDiagnostics(analyzer, actual)));
                 }
+            }
+            else
+            {
+                VerifyDiagnosticLocation(analyzer, actual, actual.Location, expected.Locations.First());
+                var additionalLocations = actual.AdditionalLocations.ToArray();
 
-                if (actual.Severity != expected.Severity)
+                if (additionalLocations.Length != expected.Locations.Length - 1)
                 {
                     Assert.True(false,
-                        string.Format("Expected diagnostic severity to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
-                            expected.Severity, actual.Severity, FormatDiagnostics(analyzer, actual)));
+                        string.Format("Expected {0} additional locations but got {1} for Diagnostic:\r\n    {2}\r\n",
+                            expected.Locations.Length - 1, additionalLocations.Length,
+                            FormatDiagnostics(analyzer, actual)));
                 }
 
-                if (actual.GetMessage() != expected.Message)
+                for (int j = 0; j < additionalLocations.Length; ++j)
                 {
-                    Assert.True(false,
-                        string.Format("Expected diagnostic message to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
-                            expected.Message, actual.GetMessage(), FormatDiagnostics(analyzer, actual)));
+                    VerifyDiagnosticLocation(analyzer, actual, additionalLocations[j], expected.Locations[j + 1]);
                 }
             }
+
+            if (actual.Id != expected.Id)
+            {
+                Assert.True(false,
+                    string.Format("Expected diagnostic id to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
+                        expected.Id, actual.Id, FormatDiagnostics(analyzer, actual)));
+            }
+
+            if (actual.Severity != expected.Severity)
+            {
+                Assert.True(false,
+                    string.Format("Expected diagnostic severity to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
+                        expected.Severity, actual.Severity, FormatDiagnostics(analyzer, actual)));
+            }
+
+            if (actual.GetMessage() != expected.Message)
+            {
+                Assert.True(false,
+                    string.Format("Expected diagnostic message to be \"{0}\" was \"{1}\"\r\n\r\nDiagnostic:\r\n    {2}\r\n",
+                        expected.Message, actual.GetMessage(), FormatDiagnostics(analyzer, actual)));
+            }
         }
```

Comment:
Minor point, but why not verify the Id, Severity and Message before trying to figure out if the locations match.